Improve rendering of pull request template in GitHub

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Adds a checklist item as a reminder to bump the version if appropriate.
- Adds checklist items as reminders to create a pre-release and a release if appropriate.
- Improves the rendering of the pull request template in GitHub.

See also cisagov/action-lineage#82.

## 💭 Motivation and context ##

- Now that all of our projects have a version associated with them it is important to have a reminder to bump the version when appropriate.
- With the version bump reminder it makes sense to have these reminders as well.
- The list items no longer appear artificially shortened and instead can be rendered as markdown sees fit.  This is something I have been correcting manually in my PRs for years.

## 🧪 Testing ##

All automated tests pass.  I also viewed [the updated version of `pull_request_template.md`](https://github.com/cisagov/.github/blob/improvement/improve-rendering-in-github/.github/pull_request_template.md) with my ocular orbs and verified that it appears correct.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Bump the patch version.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

jsf9k committed:
Improve rendering of pull request template in GitHub

All that is changed is that list items are no longer broken up into
multiple lines and a markdownlint ignore directive is added for one
particularly long markdown line.  This improves rendering in GitHub
because the list items no longer appear artificially shortened and
instead can be rendered as markdown sees fit.

jsf9k committed:
Add a checklist item for bumping the version

Now that all of our projects have a version associated with them it is
important to have a reminder to bump the version when appropriate.

jsf9k committed:
Bump version from 1.0.0 to 1.0.1

jsf9k committed:
Add checklist items to create a pre-release and a release

With the addition of the version bump checklist item these are also
necessary.

dav3r reviewed (approved):
👍 👍

mcdonnnj commented on .github/pull_request_template.md:
I think that instead of adding `<!-- markdownlint-disable-line MD013 -->` to each long line we should disable the rule for the entire file. Given the nature of the issue we cannot reasonably enforce line-length rules without impacting rendering.

jsf9k replied:
Please see commit eedadcc.

mcdonnnj reviewed (approved):
Makes sense. I dug into it a bit because the nature of the problem bothered me and found this comment https://github.com/github/markup/issues/1050#issuecomment-294654762 which touched on why rendering in the pull request behaves differently.

jsf9k committed:
Disable the markdownlint check for overly long lines globally

GitHub renders PRs such that soft line breaks are treated as hard line
breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  Hence it makes sense to disable that
rule throughout the file.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>
